### PR TITLE
Fixed selector|value-no-vendor-prefix performance

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

https://github.com/stylelint/stylelint/issues/6869

> Is there anything in the PR that needs further explanation?

selector-no-vendor-prefix

Mean: 246.52507291666666 ms
Deviation: 22.75678614753106 ms

To

Mean: 91.62227779999999 ms
Deviation: 7.8451629248369095 ms

value-no-vendor-prefix

Mean: 102.0994388 ms
Deviation: 11.78154094781758 ms

To

Mean: 91.27013606666667 ms
Deviation: 15.351019516794766 ms

### CODE REVIEW

File: .changeset/short-onions-play.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `selector|value-no-vendor-prefix` performance
```
Comment (ybiquitous) [inline .changeset/short-onions-play.md:5]: [suggest] How about a glob-like pattern? It's minor, though. 😅 

```suggestion
Fixed: `{selector,value}-no-vendor-prefix` performance
```

File: lib/rules/selector-no-vendor-prefix/index.js [hunk 2]
```diff
@@ -41,6 +42,8 @@ const rule = (primary, secondaryOptions, context) => {
 		}
 
 		root.walkRules((ruleNode) => {
+			if (!hasPrefix(ruleNode?.selector)) return;
```
Comment (ybiquitous) [inline lib/rules/selector-no-vendor-prefix/index.js:45]: [suggest] Is `ruleNode` always truthy, right?

```suggestion
			if (!hasPrefix(ruleNode.selector)) return;
```

File: lib/utils/hasPrefix.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const { prefixes } = require('../reference/prefixes');
+
+const HAS_PREFIX_REGEX = new RegExp(`(?:${[...prefixes.values()].join('|')})`, 'i');
```
Comment (ybiquitous) [inline lib/utils/hasPrefix.js:5]: [ask] Is omitting `^` (starts with) intentional, right? Acceptable?

Comment (jeddy3) [reply lib/utils/hasPrefix.js:5]: We may test broader strings, e.g. a selector list where the first selector isn't prefixed.

Comment (ybiquitous) [reply lib/utils/hasPrefix.js:5]: I see. I'm a bit concerned that `value-no-vendor-prefix` is using `startsWith`. 🤔

Comment (ybiquitous) [reply lib/utils/hasPrefix.js:5]: Ah, I misunderstood! `value-no-vendor-prefix` checks a value by `isAutoprefixable.propertyValue()`. Okay, no problem. 👍🏼

**Review by ybiquitous — APPROVED** [review]
Great! LGTM 👍🏼

File: lib/utils/hasPrefix.js (hunk 3 above)
Comment (ybiquitous) [inline lib/utils/hasPrefix.js:5]: [suggest] Sorry, I found it after approve. 😅 We can omit `.values()`:

```suggestion
const HAS_PREFIX_REGEX = new RegExp(`(?:${[...prefixes].join('|')})`, 'i');
```

**Review by ybiquitous — APPROVED** [review]
Thank you! 👍🏼

File: lib/reference/prefixes.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const prefixes = new Set(['-webkit-', '-moz-', '-ms-', '-o-']);
```
Comment (Mouvedia) [inline lib/reference/prefixes.js:1]: If this is to be reused, Ill eventually add some more rare ones to the set.

ref: https://stackoverflow.com/questions/11705144/webkit-and-its-legacy-prefixes

File: lib/rules/value-no-vendor-prefix/index.js [hunk 5]
```diff
@@ -59,11 +48,9 @@ const rule = (primary, secondaryOptions, context) => {
 		root.walkDecls((decl) => {
 			const { value } = decl;
 
-			if (
-				!isStandardSyntaxDeclaration(decl) ||
-				!isStandardSyntaxProperty(decl.prop) ||
-				!value.startsWith('-')
-			) {
+			if (!hasPrefix(value)) return;
```
Comment (Mouvedia) [inline lib/rules/value-no-vendor-prefix/index.js:51]: A more agnostic way would be to use
https://github.com/stylelint/stylelint/blob/3065425dc94be536dcf4369dcc1ed47735259e07/lib/utils/vendor.js#L22-L30
instead.

**Pull request merged**